### DISCUSSION TITLE
unable to access /admin in production Hugo + netlify

### DISCUSSION THREAD

**Discussion by elax46** [tinacms/tinacms#3859] [Help] [answered]
Hi all I have done all the steps to set up TinaCMS on Hugo and deployed via netlify. All the local procedure was successful, it is no coincidence that I can access the TinaCMS login page. During the installation I also entered the Client ID and Token for Tina Cloud but once the site was deployed going to the admin page a 404 error comes out. All the variables seem to be set correctly. Does anyone have any suggestions? I am attaching some screenshots
![Screenshot 2023-04-25 115732](https://user-images.githubusercontent.com/55278049/234242806-8ca1e5b7-7fa2-4715-8e04-716e0154faf9.png)
![Screenshot 2023-04-25 115747](https://user-images.githubusercontent.com/55278049/234242842-f4f51ba8-8831-4c8f-ad4d-733c92c7217e.png)
![Screenshot 2023-04-25 115834](https://user-images.githubusercontent.com/55278049/234242855-c39fdb3f-efd5-4404-9593-238489a8a6b6.png)

**Comment by elax46** [comment, marked as answer]
I managed to solve the problem for all those who have set the `netlify.toml` remember to modify the build command if used as if modified the one from the netlify interface will build the site with the internal command to the file and not the one set also to start TinaCMS

**Reply by jamespohalloran** [reply]
Thanks for the update. We have a note on this in our [generic Netlify doc](https://tina.io/docs/tina-cloud/connecting-site/#build-configuration), but not our framework-specific guides. I'll update that now.

**Reply by elax46** [reply]
In my opinion it is important to emphasize that environment variables should be kept private. The wizard makes them visible in the `config.ts` but I would add a paragraph specifying that for security reasons they should be kept private by replacing them with

```
   clientId: process.env.TINA_CLIENT_ID!,
   branch: process.env.TINA_BRANCH!,
   tokens: process.env.TINA_TOKEN!,
```

and by setting the variables in the respective construction services, there is a risk of leaving them public even for simple distraction
